midend/local_copyprop.cpp

```diff
@@ -409,12 +441,19 @@ IR::MethodCallExpression *DoLocalCopyPropagation::postorder(IR::MethodCallExpres
         if (actions.count(fn->path->name)) {
             LOG3("action method call " << mc->method);
             apply_function(&actions[fn->path->name]);
+            return mc;
+        } else if (fn->path->name == "verify" || fn->path->name == "hash") {
```

Comment:
Hash is not part of the standard, you should not hardwire its name here. You may want another annotation on methods to indicate that they have no side-effects. E.g. `@pure`